### PR TITLE
minor config schema changes, update getPathAnnotations and constants for new data-processing Path Annotation ETL

### PR DESCRIPTION
**Pull request by druvpatel**
- update config schema for regional annotation ETL, comment out unused schema arguments for revisit
- update TABLE_NAME in constants.py to accept "" as DATASET_NAME
- have API pull from new table directory schema from data_processing annotation ETL.
- Allow slide id or hobi input from getPathologyAnnotations endpoint

### CODE REVIEW

**Review by doori — APPROVED** [review]
Nice extension of the API, config and constants methods!

File: data_processing/common/constants.py [hunk 1]
```diff
@@ -30,9 +30,18 @@
 APP_CFG				='APP_CFG'
 DATA_CFG			='DATA_CFG'
 SCHEMA_FILE			='data_ingestion_template_schema.yml'
-
 PUBLIC_DIR          ='/gpfs/mskmind_ess/mind_public'
 
+# Annotation API Data Mappings
+PROJECT_MAPPING = {'msk_mind_ov': "OV_16-158", "msk_mind_lung": "LUNG_18-193"}
```
Comment (arfathpasha) [inline data_processing/common/constants.py:36]: I think this is OK for now but we should think of a more configurable way of specifying this mapping. This amounts to hard coding.

File: data_processing/common/constants.py [hunk 2]
```diff
@@ -30,9 +30,18 @@
...
+ANNOTATION_TABLE_MAPPINGS = \
+    {"regional": \
+        {"DATA_TYPE":"REGIONAL_CONCAT_GEOJSON", "GEOJSON_COLUMN_NAME": "concat_geojson"}, \
```
Comment (arfathpasha) [inline data_processing/common/constants.py:39]: make all same case (upper or lower) for uniformity?

File: data_processing/getPathologyAnnotations/config.yaml.template [hunk 3]
```diff
@@ -1,3 +1,27 @@
+spark_cluster_config:
```
Comment (arfathpasha) [inline data_processing/getPathologyAnnotations/config.yaml.template:1]: I suggest we use the terms app_config and data_config (previously ingestion_config). So this file name would become app_config.yaml.template to further qualify it as a config for the app.

**Review by arfathpasha — APPROVED** [review]
Few minor comments.

File: data_processing/common/constants.py (hunk 1 above)
Comment (druvpatel) [reply data_processing/common/constants.py:36]: Right -- maybe a YAML here would work

**Pull request merged**